src/vs/workbench/contrib/extensions/browser/extensionsWorkbenchService.ts

```diff
@@ -949,10 +951,25 @@ export class ExtensionsWorkbenchService extends Disposable implements IExtension
 		}));
 
 		urlService.registerHandler(this);
+		this.registerConfigurations();
 
 		this.whenInitialized = this.initialize();
 	}
 
+	private registerConfigurations(): void {
+		Registry.as<IConfigurationRegistry>(ConfigurationExtensions.Configuration)
+			.registerConfiguration({
+				...extensionsConfigurationNodeBase,
+				properties: {
+					[AutoRestartConfigurationKey]: {
+						type: 'boolean',
+						description: nls.localize('autoRestart', "If activated, extensions will automatically restart following an update if the window is not in focus."),
+						default: this.productService.quality !== 'stable',
```

Comment:
Should have a follow up to enable this by default independent from quality and then move this into a static call (contribution file), not as part of the service.

Reply:
Modifying the configuration at a later stage necessitates remembering the specific configuration node to update. However, for this particular setting, making such an adjustment later isn't justified because registering this configuration subsequently is acceptable. The only feasible action is to shift this registration to a static contribution.